src/txn.ts

```diff
@@ -31,13 +31,21 @@ export class Txn {
     private readonly ctx: messages.TxnContext;
     private finished: boolean = false;
     private mutated: boolean = false;
+    private readonly readOnly: boolean = false;
+    private useBestEffort: boolean = false;
     private sequencingProp: messages.LinRead.SequencingMap[keyof messages.LinRead.SequencingMap];
 
-    constructor(dc: DgraphClient) {
+    constructor(dc: DgraphClient, txnOpts?: { readOnly: boolean } | null) {
         this.dc = dc;
         this.ctx = new messages.TxnContext();
         this.ctx.setLinRead(this.dc.getLinRead());
         this.sequencingProp = messages.LinRead.Sequencing.CLIENT_SIDE;
+        this.readOnly = txnOpts !== undefined ? txnOpts.readOnly : false;
+    }
+
+    public bestEffort(): this {
```

Comment:
You are passing readOnly as `txnOptions` but bestEffort() follows a "builder" pattern.
I think it'll be more consistent to just pass both flags in the `txnOptions` object.

Reply:
@paulftw I asked @danielmai about this and we were trying to follow the same structure as `dgo`. Up to you.

`newReadOnlyTxn().BestEffort()` is what the public API looks like for `dgo`. IDC either way

Reply:
I don't want to publicly criticize dgo choices, but seems like I already did :) let's put everything in options.

Reply:
Haha, okay sounds good.